app/models/topic.rb

```diff
@@ -25,7 +26,43 @@
 class Topic < Tag
   has_many :mainstream_browse_pages, through: :reverse_tag_associations, source: :from_tag
 
+  alias subtopic? has_parent?
+
   def legacy_tag_type
     'specialist_sector'
   end
+
+  # Define redirect routes by returning an array of hashes:
+  #
+  # `{ '/foo' => { '/foo' => '/bar' } }` will add a redirect route to the
+  # content-item, causing gov.uk/foo to redirect to gov.uk/bar
+  def redirects
+    redirect_to_topic_namespace
+  end
+
+  def base_path
+    "/topic#{super}"
+  end
+
+private
+
+  # Redirects are currently hard-coded: if this topic existed before the move
+  # to the /topic namespace, we add a redirect from the old path (/oil-and-gas)
+  # to the namespaced version (/topic/oil-and-gas).
+  # For subtopics we also redirect two subroutes:
+  # /:topic/:subtopic/email-signup and /:topic/:subtopic/latest
+  def redirect_to_topic_namespace
+    return {} unless has_legacy_route_without_namespace?
+
+    old_path = base_path.gsub('/topic', '')
```

Comment:
Worth an anchor in this regex.  We wouldn't want this to hit any false-positives (I'm imagining a topic with a slug of "topical-events" or similar).

Also, this only needs to be `sub` not `gsub` as it should only be making a single replacement.